tests/checks/git.fish

```diff
@@ -46,6 +46,8 @@ echo
 #CHECK: (newbranch|…1)
 set -e __fish_git_prompt_show_informative_status
 
+set -e __fish_git_prompt_show_informative_status
```

Comment:
Huh deleting it twice won't trigger another handler, right? Unless there is a spooky universal variables around...

(Re the commit message: the variable handler behavior for variable overrides seems correct as far as I can tell)

Reply:
Oh this is a rebase artifact (I did this for #7926 as well).

>(Re the commit message: the variable handler behavior for variable overrides seems correct as far as I can tell)

The problem is that the handler only triggers when the override goes *into* scope, not also when it goes *out* of scope. So since we disable informative mode stuff when the variable is erased, we don't notice when it's used as an override.

See:

```fish
> set -l foo bar
> function on-foo --on-variable foo; echo on foo: $argv; echo foo is $foo; end
> foo=banana echo bar
on foo: VARIABLE SET foo
foo is banana
bar
```

See how the handler is never triggered for foo being "bar" *again*? That's the problem here.

Reply:
I'm not entirely sure how this *should* act, just saying that it gets in the way here.

Reply:
That sounds correct, because the override acts just like `begin; set -l;  ...; end`

Reply:
That's *consistent*, I'm not sure it's *correct*.

Maybe we should fire the variable handlers when it goes out of scope? Or would that massively slow down block closing?

Reply:
I guess this problem has just never come up before because variable handlers are primarily used on global/universal variables.
Firing the handlers seems useful here - for running the git prompt with different configurations via variable overrides.
For this use case it's enough to fire handlers when a variable, that shadows another variables, goes out of scope; not when a local-only variable dies.

(Ideally we don't need the caching logic of `___fish_git_prompt_init`, I'm assuming it's a sound optimization).